cli_tools_e2e_test/gce_image_import_export/test_suites/onestep_import/onestep_import_test_suite.go

```diff
@@ -0,0 +1,246 @@
+}
+
+func runOnestepImageImportFromAWSWindowsAMI(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger,
+	testProjectConfig *testconfig.Project, testType utils.CLITestType) {
+	imageName := "e2e-test-onestep-image-import" + path.RandString(5)
+
+	props := &onestepImportAWSTestProperties{
+		imageName:     imageName,
+		amiID:         windowsAMIID,
+		os:            "windows-2019",
+		timeout:       "4h",
+		startupScript: "post_translate_test.ps1",
+	}
+
+	runOnestepImportTest(ctx, props, testProjectConfig, testType, logger, testCase)
+}
+
+func runOnestepImageImportFromAWSWindowsVMDK(ctx context.Context, testCase *junitxml.TestCase, logger *log.Logger,
+	testProjectConfig *testconfig.Project, testType utils.CLITestType) {
+	imageName := "e2e-test-onestep-image-import" + path.RandString(5)
+
+	props := &onestepImportAWSTestProperties{
+		imageName:         imageName,
+		sourceAMIFilePath: windowsVMDKFilePath,
+		os:                "windows-2019",
+		startupScript:     "post_translate_test.ps1",
+	}
+
+	runOnestepImportTest(ctx, props, testProjectConfig, testType, logger, testCase)
+}
+
+func runOnestepImportTest(ctx context.Context, props *onestepImportAWSTestProperties, testConfig *testconfig.Project, testType utils.CLITestType,
+	logger *log.Logger, testCase *junitxml.TestCase) {
+	args := buildTestArgs(props, testConfig)[testType]
+
+	cmds := map[utils.CLITestType]string{
+		utils.Wrapper:                   "./gce_onestep_image_import",
+		utils.GcloudProdWrapperLatest:   "gcloud",
+		utils.GcloudLatestWrapperLatest: "gcloud",
+	}
+
+	if utils.RunTestForTestType(cmds[testType], args, testType, logger, testCase) {
+		verifyImportedImageFile(ctx, testCase, props, testConfig, logger)
+	}
+}
+
+// buildTestArgs build args for tests.
+func buildTestArgs(props *onestepImportAWSTestProperties, testProjectConfig *testconfig.Project) map[utils.CLITestType][]string {
+	gcloudArgs := []string{
+		"beta", "compute", "images", "import", "--quiet",
+		"--docker-image-tag=latest", props.imageName,
+		fmt.Sprintf("--project=%v", testProjectConfig.TestProjectID),
+		fmt.Sprintf("--aws-access-key-id=%v", awsAccessKeyID),
+		fmt.Sprintf("--aws-secret-access-key=%v", awsSecretAccessKey),
+		fmt.Sprintf("--aws-session-token=%v", awsSessionToken),
+		fmt.Sprintf("--aws-region=%v", awsRegion),
+		fmt.Sprintf("--os=%v", props.os),
+	}
+
+	wrapperArgs := []string{
+		"-client_id=e2e",
+		fmt.Sprintf("-image_name=%v", props.imageName),
+		fmt.Sprintf("-project=%v", testProjectConfig.TestProjectID),
+		fmt.Sprintf("-aws_access_key_id=%v", awsAccessKeyID),
+		fmt.Sprintf("-aws_secret_access_key=%v", awsSecretAccessKey),
+		fmt.Sprintf("-aws_session_token=%v", awsSessionToken),
+		fmt.Sprintf("-aws_region=%v", awsRegion),
+		fmt.Sprintf("-os=%v", props.os),
+	}
+
+	if props.amiID != "" {
+		gcloudArgs = append(gcloudArgs, fmt.Sprintf("--aws-ami-export-location=%v", awsBucket))
```

Comment:
I understand you want to reduce dup code. However, it's hard to read and fix a test. I suggest to put the arg logic directly in the test case.